### PR TITLE
Add support for ghc-9.0.1

### LINKED ISSUE #7: Can't build with ghc-9.0.1
```
[1 of 1] Compiling Language.Haskell.TH.Extras ( src/Language/Haskell/TH/Extras.hs, /tmp/th-extras/dist-newstyle/build/x86_64-linux/ghc-9.0.1/th-extras-0.0.0.4/build/Language/Haskell/TH/Extras.o, /tmp/th-extras/dist-newstyle/build/x86_64-linux/ghc-9.0.1/th-extras-0.0.0.4/build/Language/Haskell/TH/Extras.dyn_o )

src/Language/Haskell/TH/Extras.hs:49:17: error:
    • Expecting one more argument to ‘TyVarBndr’
      Expected a type, but ‘TyVarBndr’ has kind ‘* -> *’
    • In the type signature: nameOfBinder :: TyVarBndr -> Name
   |
49 | nameOfBinder :: TyVarBndr -> Name
   |                 ^^^^^^^^^

src/Language/Haskell/TH/Extras.hs:58:27: error:
    • Expecting one more argument to ‘TyVarBndr’
      Expected a type, but ‘TyVarBndr’ has kind ‘* -> *’
    • In the type signature: varsBoundInCon :: Con -> [TyVarBndr]
   |
58 | varsBoundInCon :: Con -> [TyVarBndr]
   |                           ^^^^^^^^^

src/Language/Haskell/TH/Extras.hs:254:28: error:
    • Expecting one more argument to ‘TyVarBndr’
      Expected a type, but ‘TyVarBndr’ has kind ‘* -> *’
    • In the first argument of ‘Q’, namely ‘([TyVarBndr], Int)’
      In the type signature: tyConArity' :: Name -> Q ([TyVarBndr], Int)
    |
254 | tyConArity' :: Name -> Q ([TyVarBndr], Int)
    |                            ^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by anka-213**
Fixes #7

### CODE REVIEW

File: th-extras.cabal [hunk 1]
```diff
@@ -34,7 +34,8 @@ Library
   exposed-modules:      Language.Haskell.TH.Extras
   build-depends:        base >= 3 && < 5,
                         containers,
-                        template-haskell
-  
+                        template-haskell,
```
Comment (phadej) [inline th-extras.cabal:37]: i'd recommend upper bounds. `template-haskell` changes with every GHC.

Comment (anka-213) [reply th-extras.cabal]: Good point

**Comment by jneira** [comment]
Hi, @ali-abrar  is there some chance to add support for ghc-9 with this pr? thanks in advance!

**Comment by pepeiborra** [comment]
Yet another friendly reminder that this is blocking HLS from cabal installing with ghc 9.0

**Comment by jneira** [comment]
@mokus0 hi! any chance to have some time to review and merge this? thanks!!

**Comment by ryantrinkle** [comment]
@jneira Thanks for emailing!  I hadn't seen this.

**Comment by jneira** [comment]
@ryantrinkle @mokus0 thanks for take care, any chance to complete the merge with a hackage release? many thanks!

**Pull request merged**